lib/listener-removed.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const { EventListener } = require('@janiscommerce/event-listener');
+const { ApiSession } = require('@janiscommerce/api-session');
+const Model = require('@janiscommerce/model');
+
+const ModelGetter = require('./helper/model-getter');
+
+module.exports = class ClientRemovedListener extends EventListener {
+
+	get mustHaveClient() {
+		return true;
+	}
+
+	get model() {
+		return ModelGetter.getInstance();
+	}
+
+	async process() {
+
+		const code = this.eventClient;
+
+		const clientToDrop = await this.model.getBy('code', code, { unique: true });
+
+		if(!clientToDrop) {
+			this.setCode(404);
```

Comment:
```suggestion
this.setCode(404)
	.setBody('Client not found in Service');
```
Es posible hacer esto por es  _chainable_, es decir... cada método devuelve la instancia luego de realizar su acción